packages/engine/Source/Scene/I3SField.js

```diff
@@ -252,4 +272,109 @@ I3SField.prototype._parseBody = function (dataView, offset) {
   }
 };
 
+/**
+ * @private
+ */
+I3SField.prototype._getBodyOffset = function (headerSize) {
+  let valueSize = 0;
+  if (defined(this._storageInfo.attributeValues)) {
+    valueSize = getNumericTypeSize(this._storageInfo.attributeValues.valueType);
+  } else if (defined(this._storageInfo.objectIds)) {
+    valueSize = getNumericTypeSize(this._storageInfo.objectIds.valueType);
+  }
+  if (valueSize > 0) {
+    // Values will be padded to align the addresses with the data size
+    return Math.ceil(headerSize / valueSize) * valueSize;
+  }
+  return headerSize;
+};
+
+/**
+ * @private
+ */
+I3SField.prototype._validateHeader = function (dataView) {
+  let headerSize = 0;
+  for (
+    let itemIndex = 0;
+    itemIndex < this._storageInfo.header.length;
+    itemIndex++
+  ) {
+    const item = this._storageInfo.header[itemIndex];
+    headerSize += getValueTypeSize(item.valueType);
+  }
+  if (dataView.byteLength < headerSize) {
+    console.error("Invalid attribute buffer size");
```

Comment:
This case should throw an exception rather than logging to the console.

Reply:
And instead of returning `true` or `false`, when this function is invoked, tit should be surrounded by a `try/catch` block.

Reply:
Done.